pandas/core/strings.py

```diff
@@ -344,17 +351,28 @@ def str_replace(arr, pat, repl, n=-1, case=None, flags=0):
     Notes
     -----
     When `pat` is a compiled regex, all flags should be included in the
-    compiled regex. Use of `case` or `flags` with a compiled regex will
-    raise an error.
+    compiled regex. Use of `case`, `flags`, or `regex` with a compiled regex
```

Comment:
this contradicting the below. regex=True *is* compatible with a compiled regex no?

Reply:
My thought was that specifying nondefault `regex` is incompatible (same as the others), but agree it sounds weird since regex defaults True. I'll change to "Use of case, flags, or regex=False" which hopefully should be clearer.